Migrate to DTMT

sclu1034 opened the pull request:
I guess one could invoke https://xkcd.com/1172/ for the complete removal of I/O.

sclu1034 committed:
feat: Make buildable with dtmt

Move files in the correct file structure, add package definition and
dtmt configuration.

sclu1034 committed:
refactor: Replace I/O-based requires

With the switch back to bundle-based mod distribution, we can now use
the built-in `require`/`dofile`.

sclu1034 committed:
chore: Add more globals to luacheckrc

sclu1034 committed:
fix: Adjust for adapted mod manager

sclu1034 committed:
fix: Change mod ID to match usage in code

sclu1034 committed:
fix: Fix script file paths

sclu1034 committed:
feat: Add stack trace to argument check error

Just the error message is not enough to track errors in commonly used
functions.

sclu1034 committed:
feat: Improve error message logging

sclu1034 committed:
refactor: Use a proper API to get mod manager data

sclu1034 committed:
chore: Update config with new fields

danreeves commented on scripts/mods/dmf/modules/core/safe_calls.lua:
```diff
@@ -14,9 +14,14 @@ end
   if type(error_message) == "table" and error_message.error then
-    error_message = error_message.error
+    print(string.format(
+        "<<Script Error>>%s<<Script Error>>\n<<Lua Stack>>%s<</Lua Stack>>\n<<Lua Locals>>%s<</Lua Locals>>\n<<Lua Self>>%s<</Lua Self>>",
```
```suggestion
        "<<Script Error>>%s<</Script Error>>\n<<Lua Stack>>%s<</Lua Stack>>\n<<Lua Locals>>%s<</Lua Locals>>\n<<Lua Self>>%s<</Lua Self>>",
```

sclu1034 committed:
Fix error print format

Co-authored-by: Dan Reeves <hey@danreev.es>

sclu1034 committed:
Merge remote-tracking branch 'upstream/master' into feat/mod-bundles

* upstream/master:
  Add parameter defaults and better error messages for dump and dump_to_file (#37)

sclu1034 committed:
Ignore build directory

sclu1034 committed:
Fix mod options not loading

The initialization for the options view happened before a few necessary
systems where setup by the game.

sclu1034 committed:
fix: Fix mod options after reload

sclu1034 committed:
Format luacheckrc

sclu1034 committed:
Implement loading non-bundled mods
